src/selection-handler.js

```diff
@@ -164,16 +202,116 @@ Object.assign(SelectionHandler.prototype, require('./function-bind'), require('.
+    }
+    return copyArea;
+  },
+
   delete: function() {
     this.editor.getBlocks().forEach((blockEl, idx) => {
       if (this.indexSelected(idx)) this.mediator.trigger("block:remove", block.blockID, { focusOnNext: true });
     });
   },
 
   indexSelected: function(index) {
-    return index >= Math.min(this.startIndex, this.endIndex) && index <= Math.max(this.startIndex, this.endIndex);
-  }
+    return index >= this.getStartIndex() && index <= this.getEndIndex();
+  },
+
+  block: function(block) {
+    var blockPosition = this.editor.blockManager.getBlockPosition(block.el);
+
+    this.mediator.trigger("formatter:hide");
+    this.removeNativeSelection();
+    this.start(blockPosition);
+  },
+
+  getStartIndex: function() {
+    return Math.min(this.startIndex, this.endIndex);
+  },
+
+
+  getEndIndex: function() {
+    return Math.max(this.startIndex, this.endIndex);
+  },
+
+  getStartBlock: function() {
+    return this.editor.getBlocks()[this.getStartIndex()];
+  },
 
+  getEndBlock: function() {
+    return this.editor.getBlocks()[this.getEndIndex()];
+  },
+
+  onKeyDown: function(ev) {
+    ev = ev || window.event;
+    var ctrlKey = ev.ctrlKey || ev.metaKey;
+    var key = ev.key;
+
+    if (ctrlKey && key === "a") {
+      if (!this.selecting && this.cancelSelectAll()) return;
+      ev.preventDefault();
+      this.mediator.trigger("selection:all");
+    } else if (this.selecting && ["Down", "ArrowDown"].indexOf(key) > -1) {
+      ev.preventDefault();
+      if (ev.shiftKey && ev.altKey) this.expandToEnd();
+      else if (ev.shiftKey) this.expand(1);
+      else if (ev.altKey) this.startAtEnd();
+      else {
+        this.cancel();
+        this.mediator.trigger("block:focusNext", this.getEndBlock().blockID, { force: true });
+        return;
+      }
+      this.focusAtEnd();
+    } else if (this.selecting && ["Up", "ArrowUp"].indexOf(key) > -1) {
+      ev.preventDefault();
+      if (ev.shiftKey && ev.altKey) this.expandToStart();
+      else if (ev.shiftKey) this.expand(-1);
+      else if (ev.altKey) this.start(0);
+      else {
+        this.cancel();
+        this.mediator.trigger("block:focusPrevious", this.getStartBlock().blockID, { force: true });
+        return;
+      }
+      this.focusAtEnd();
+    }
+  },
+
+  onMouseUp: function() {
+    if (!window.mouseDown) {
+      this.mediator.trigger("selection:complete");
+      this.mediator.trigger("selection:cancel");
+      return;
+    }
+
+    window.mouseDown = false;
+    this.mediator.trigger("selection:complete");
+  },
+
+  onCopy: function(ev) {
+    if (!this.selecting) return;
+
+    var content = this.getClipboardData();
+
+    ev.clipboardData.setData(TYPE, JSON.stringify(content.data));
+    ev.clipboardData.setData('text/html', content.html);
+    ev.clipboardData.setData('text/plain', content.text);
+    ev.preventDefault();
+  },
+
+  onCut: function(ev) {
+    this.delete();
+  },
+
+  onPaste: function(ev) {
+    // TODO
```

Comment:
Is this PR a WIP then?

Reply:
I put that in place so I could add the config all at once.